components/list/list-item-mixin.js

```diff
@@ -341,18 +388,18 @@ export const ListItemMixin = superclass => class extends ListItemDragDropMixin(L
 			'd2l-list-item-content-extend-separators': this._extendSeparators,
 			'd2l-focusing': this._focusing,
 			'd2l-hovering': this._hovering,
+			'd2l-dragging-over': this._draggingOver
```

Comment:
Yeah I hear ya. We tried pretty hard to keep things separated, but ultimately some of these things need to be aware of each other.  In this case the `d2l-list-item-generic-layout` needs to know when the user is dragging over so that it can stack the nested list on top of the drop target layer of the parent list-item (otherwise their `dragstart` events will never fire).

I could move this class to a method on the drag & drop mixin - say `getDragDropListItemClasses()`.  That would improve the leakiness slightly.  Thoughts on that?  Either way, the `d2l-list-item-generic-layout` still needs to be aware of the class, or `ListItemDragDropMixin` needs to be aware of `d2l-list-item-generic-layout`. I chose to keep all the stacking rules together, in `d2l-list-item-generic-layout` since its much easier to understand what's going on when they are located in the same place.

Reply:
This is clear enough what's going on, and yeah they're all pretty tightly coupled together anyway. I'm good either way.